### DISCUSSION TITLE
Large build won't be precached

### DISCUSSION THREAD

**Discussion by JacobBlana** [outline/outline#3527] [Self Hosting]
Hi, i made the self hosted outline working but now I am left with warning like this:

```
[Log] ./node_modules/@renderlesskit/react/dist/esm/drawer/index.js 3:0-263
/static/main.e84a5959efe9aa6d782e.js is 15.9 MB, and won't be precached. Configure maximumFileSizeToCacheInBytes to change this limit.
```

I get that I can probably just increase the "maximumFileSizeToCacheInBytes" and call it a day. But is it OK in the end that the build is that large? Thanks

**Comment by gekomod** [comment]
find webpack.config.js
Edit
```code
maximumFileSizeToCacheInBytes: 15 * 1024 * 1024, // For large bundles
```

Replace  to

```code
maximumFileSizeToCacheInBytes: 17 * 1024 * 1024, // For large bundles
```

This resolve Error.
For me, this change worked correctly

**Comment by tommoor** [comment]
I suspect you're running in development mode, the largest chunk in production is 1.3Mb. You should use the prebuilt docker images to avoid issues like this.

**Comment by JacobBlana** [comment]
@tommoor we are building the app with "yarn build" as refered in the docs.. We kinda wanted to go without docker and everything seems to work kinda fine. Any idea how to build only for production ? it doesnt look like there is different build command for it.. thanks

**Reply by almereyda** [reply]
You could use the Dockerfiles as an example on how to set up the build process for Outline. From quickly looking at them, it appears the `--frozen-lockfile` flag will omit installing `devDependencies`, despite `NODE_ENV` is not set in the image(s), and thus left at the `development` default. For reference, which command do you use for installing the dependencies into your custom build?

In https://app.getoutline.com/share/770a97da-13e5-401e-9f8a-37949c19f97e/doc/from-source-BlBxrNzMIP they also add a parameter `--production=true` to the `install` command, prior running `build`.

You could also try to set `NODE_ENV=production` and see if that changes anything for your `yarn build`.
